### DISCUSSION TITLE
Feature request: Add annotation(s) into the legend

### DISCUSSION THREAD

**Discussion by HuongSakada** [chartjs/chartjs-plugin-annotation#418] [Ideas]
For Chartjs, we can create the chart with the legend and can show or hide it by selecting the legend.
Do we have any ways to add the annotation into the legend?
<img width="1085" alt="Screen Shot 2021-01-20 at 3 26 58 PM" src="https://user-images.githubusercontent.com/22464193/105148046-8011a880-5b34-11eb-8ea9-c14cce95cee1.png">

**Comment by stockiNail** [comment]
@HuongSakada my personal opinion, that's not possible because legend items are representing a dataset or data of dataset. In this case you would like to have a legend item to a chart element.

I guess (but correct me if I'm wrong), you want to have the possibility to show and hide the annotation on demand.
If you are using the [last pre-release v1.0.0-beta.2](https://github.com/chartjs/chartjs-plugin-annotation/releases/tag/v1.0.0-beta.2), this is possible by `display` options and of course an external HTML "controller" which can enable or disable the display of the label.

**Comment by HuongSakada** [comment]
Yes, you are right. If we want to show the legend it must be added in datasets.
So I have a trick, add it as data in datasets of `type: 'line' `with `data: []`. I don't think a good solution but it can help.
```javascript
{
    type: 'line',
    label: 'The legend label',
    backgroundColor: 'red',
    data: []
}
```

**Comment by LeeLenaleee** [comment]
If you really want a label in your legend without adding an extra dataset as posted as a solution you can also define your own [generateLabels](https://www.chartjs.org/docs/master/configuration/legend.html#legend-label-configuration) function that returns the label items for all datasets and one extra for your custom informative label
